pkg/kv/kvserver/rangefeed/buffered_registration.go

```diff
@@ -65,6 +63,10 @@ type bufferedRegistration struct {
 		// that disconnect is called, it is closed by disconnect.
 		catchUpIter *CatchUpIterator
 	}
+
+	// Number of events that have been written to the buffer but
+	// not sent. Used only for testing.
+	testPendingEventToSend atomic.Int64
```

Comment:
nit: I'd recommend not calling this `test` since it is read by the non-test method`waitForCaughtUp`. Instead maybe just mark as internal?

Reading the code more, it appears `waitForCaughtUp` is only used in tests as well, so this is probably OK, although consider adding a comment to that method to mention that it is only used in tests. Then it would be OK to leave the variable as a `test` name.